bookkeeper-server/src/test/java/org/apache/bookkeeper/util/collections/ConcurrentLongLongPairHashMapTest.java

```diff
@@ -113,6 +113,22 @@ public void testRemove() {
         assertTrue(map.isEmpty());
     }
 
+    @Test
+    public void testExpandAndShrink() {
+        ConcurrentLongLongPairHashMap map = new ConcurrentLongLongPairHashMap(2, 1);
+        assertTrue(map.put(1, 1, 11, 11));
+        assertTrue(map.put(2, 2, 22, 22));
+        assertTrue(map.put(3, 3, 33, 33));
+
+        // expand hashmap
+        assertTrue(map.capacity() == 8);
+
+        assertTrue(map.remove(1, 1, 11, 11));
+        assertTrue(map.remove(2, 2, 22, 22));
+        // shrink hashmap
+        assertTrue(map.capacity() == 4);
```

Comment:
Can we also verify that the map is able to expand again?

Also, we should also verify that the map does not keep shrinking at every remove() operation

Here in the test case we have to ensure that the behaviour is as expected

Reply:
OK

Reply:
Fixed.

Reply:
Please review again! @eolivelli